File: 1756-design-most-recently-used-queue/1756-design-most-recently-used-queue.cpp
class MRUQueue {
    list<int> Q;
public:
    MRUQueue(int n): Q(0) {
        for(int i = 1; i <= n; i++){
            Q.push_back(i);
        }
    }
    
    int fetch(int k) {
        auto it = next(Q.begin(), k -1);
        int returnVal = *(it);
        Q.erase(it);
        Q.push_back(returnVal);
        
        return returnVal;
    }
};

/**
 * Your MRUQueue object will be instantiated and called as such:
 * MRUQueue* obj = new MRUQueue(n);
 * int param_1 = obj->fetch(k);
 */

File: 1756-design-most-recently-used-queue/1756-design-most-recently-used-queue.py
import sortedcontainers
class MRUQueue:

    def __init__(self, n: int):
        self.sd = sortedcontainers.SortedDict(
            {
                key: key for key in range(1, n + 1)
            }
        ) 
        
        self.maxKey = n + 1

    def fetch(self, k: int) -> int:
        _, value = self.sd.popitem(index=k - 1)
        
        self.sd[self.maxKey] = value
        self.maxKey += 1
        
        return value


# Your MRUQueue object will be instantiated and called as such:
# obj = MRUQueue(n)
# param_1 = obj.fetch(k)